Artifacts/windows-add-local-admin/artifact.ps1

```diff
@@ -0,0 +1,137 @@
+[CmdletBinding()]       
+param
+(
+    [Parameter(Mandatory = $true)]
+    [string] $DomainJoinUsername
+)
+
+##############################
+# Function returns the localized group name for the local administrator group
+function Get-LocalAdminGroupName ()
+{
+    #well known SID for the Administrators group
+    ([wmi]"Win32_SID.SID='S-1-5-32-544'").AccountName
+}
+
+#############################
+# Function to find the members of a local machine group, usually administrators.
+# Works around the cahnge in call pattern InvokeMember
+function Get-LocalGroupMembersPs3to5 ([Parameter(Mandatory = $true)] [string] $GroupName)
+{
+    if ($PSVersionTable.PSVersion.Major -gt 4)
+    {
+        throw "This method id not supported on powershell 5 and greater"
+    }
+    $group = [ADSI]("WinNT://$env:COMPUTERNAME/$GroupName, group");
+    $members = @($group.psbase.Invoke("Members"));
+    $Details = @();
+
+    $members | ForEach-Object {
+        $name = $_.GetType().InvokeMember("Name", 'GetProperty', $null, $_, $null)
+        $path = $_.GetType().InvokeMember("ADsPath", 'GetProperty', $null, $_, $null)
+        $schema = $_.GetType().InvokeMember("Schema", 'GetProperty', $null, $_, $null)
+       
+
+        if ($schema -notlike "WinNT://$name/*")
+        {
+            $Type = "Domain"
+            if ($path -like "*/$env:COMPUTERNAME/*")
+            {
+                $Type = "Local"
+            }
+           
+            $Details += New-Object PSObject -Property @{"Account" = $name; "Type" = $type; }
+        }
+    }
+    Write-Output "Members of $GroupName are:"
+    $Details
+}
+
+##############################
+# Function to find the members of a local machine group, usually administrators
+function Get-LocalGroupMembers ([Parameter(Mandatory = $true)] [string] $GroupName)
+{
+    if ($PSVersionTable.PSVersion.Major -lt 5)
+    {
+        return Get-LocalGroupMembersPs3to5 $GroupName
+    }
+
+    $group = [ADSI]("WinNT://$env:COMPUTERNAME/$GroupName, group");
+    $members = @($group.psbase.Invoke("Members"));
+    $Details = @();
+    
+    $members | ForEach-Object {
+        $name = $_.GetType.Invoke().InvokeMember("Name", 'GetProperty', $null, $_, $null)
+        $path = $_.GetType.Invoke().InvokeMember("ADsPath", 'GetProperty', $null, $_, $null)
+        $schema = $_.GetType.Invoke().InvokeMember("Schema", 'GetProperty', $null, $_, $null)
+
+        if ($schema -notlike "WinNT://$name/*")
+        {
+            $Type = "Domain"
+            if ($path -like "*/$env:COMPUTERNAME/*")
+            {
+                $Type = "Local"
+            }
+           
+            $Details += New-Object PSObject -Property @{"Group" = $GroupName; "Account" = $name; "Type" = $type; }
+        }
+    }
+    Write-Output "Members of $GroupName are:"
+    $Details.ForEach( {$_}) | Format-Table -Property Account, Type -AutoSize
```

Comment:
This function should only return the data and the formatting should be done in the main body where the call happens. Specially true as we have 2 functions, doing very similar work, but now formatting the output differently (i.e. `Get-LocalGroupMembersPs3to5`).